### PR TITLE
add dnd support to search panel, highlight searched rule, UI adjustments

### PR DESCRIPTION
**Pull request by yuntongf**
- Can now drag a course from search result to plan panel, req panel, dock
- When user clicks search for a specific rule, the rule will be highlighted on the req panel as the search panel displays search results
- Reduce gap between req and search panels

### CODE REVIEW

File: frontend/degree-plan/components/FourYearPlanPage.tsx
Comment (AaDalal) [inline frontend/degree-plan/components/FourYearPlanPage.tsx:107]: search state is in a [context](https://github.com/pennlabs/penn-courses/blob/9d83d4dcab84f1d16924e2b7b669879fbf68219d/frontend/degree-plan/components/Search/SearchPanel.tsx#L20) since we set it from a bunch of different places in the UI... do we also need this here?

File: frontend/degree-plan/components/Requirements/QObject.tsx
Comment (AaDalal) [inline frontend/degree-plan/components/Requirements/QObject.tsx:86]: I think this shadow looks a little weird bc it's so strong. Maybe we could change the color or apply some other effect to the surface of the search condition when clicked? let me know what you think

**Review by AaDalal — COMMENTED** [review]
couple of questions

File: frontend/degree-plan/components/FourYearPlanPage.tsx
Comment (yuntongf) [reply frontend/degree-plan/components/FourYearPlanPage.tsx:107]: changed this to using searchRuleId which already exists instead of a new state variable

File: frontend/degree-plan/components/Requirements/QObject.tsx
Comment (yuntongf) [reply frontend/degree-plan/components/Requirements/QObject.tsx:86]: I changed this to lighter and less aggressive shadow, similar to the one shown for droppable boxes

File: frontend/degree-plan/components/Search/SearchPanel.tsx
Comment (AaDalal) [inline frontend/degree-plan/components/Search/SearchPanel.tsx:106]: I think the value should be null instead of -1

**Review by AaDalal — APPROVED** [review]
LGTM except 1 comment. Just gonna approve but if you could fix that and check ui after #650 is merged that would be amazing

File: frontend/degree-plan/components/Search/SearchPanel.tsx
Comment (yuntongf) [reply frontend/degree-plan/components/Search/SearchPanel.tsx:106]: Yea makes sense

**Pull request merged**
